homeassistant/components/light/flux_led.py

```diff
@@ -253,25 +248,32 @@ def turn_on(self, **kwargs):
         if brightness is None:
             brightness = self.brightness
 
+        if hs_color and brightness:
+            self._color = color_util.color_hsv_to_RGB(hs_color[0], hs_color[1],
+                                                      brightness / 255 * 100)
+
         # Preserve color on brightness/white level change
         if rgb is None:
-            rgb = self._bulb.getRgb()
-
-        if white is None and self._mode == MODE_RGBW:
-            white = self.white_value
+            rgb = self._color
```

Comment:
When is `rgb` _not_ `None` at this point?

Reply:
Oh, my fault. I've corrected it.